src/machine/asm/execute_aarch64.S

```diff
@@ -0,0 +1,1644 @@
+  ldr TEMP2, [MACHINE, CKB_VM_ASM_ASM_CORE_MACHINE_OFFSET_CYCLES]
+  ldr TEMP1, [TRACE, CKB_VM_ASM_TRACE_OFFSET_CYCLES]
+  add TEMP2, TEMP2, TEMP1
+  cmp TEMP2, TEMP1
+  bcc .exit_cycles_overflow
+  ldr TEMP1, [MACHINE, CKB_VM_ASM_ASM_CORE_MACHINE_OFFSET_MAX_CYCLES]
+  cmp TEMP2, TEMP1
+  bhi .exit_max_cycles_exceeded
+  str TEMP2, [MACHINE, CKB_VM_ASM_ASM_CORE_MACHINE_OFFSET_CYCLES]
+  add TEMP3, TEMP3, TEMP4
+  str TEMP3, [MACHINE, CKB_VM_ASM_ASM_CORE_MACHINE_OFFSET_PC]
+  add INST_ARGS, TRACE, CKB_VM_ASM_TRACE_OFFSET_INSTRUCTIONS
+  add INST_PC, TRACE, CKB_VM_ASM_TRACE_OFFSET_THREAD
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_ADDI:
+  DECODE_I
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  add RS1, RS1, IMMEDIATE
+  WRITE_RD(RS1)
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_ADD:
+  DECODE_R
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  ldr RS2, REGISTER_ADDRESS(RS2)
+  add RS1, RS1, RS2
+  WRITE_RD(RS1)
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_ADDIW:
+  DECODE_I
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  add RS1, RS1, IMMEDIATE
+  sxtw RS1, RS1w
+  WRITE_RD(RS1)
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_ADDW:
+  DECODE_R
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  ldr RS2, REGISTER_ADDRESS(RS2)
+  add RS1, RS1, RS2
+  sxtw RS1, RS1w
+  WRITE_RD(RS1)
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_AND:
+  DECODE_R
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  ldr RS2, REGISTER_ADDRESS(RS2)
+  and RS1, RS1, RS2
+  WRITE_RD(RS1)
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_ANDI:
+  DECODE_I
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  and RS1, RS1, IMMEDIATE
+  WRITE_RD(RS1)
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_AUIPC:
+  DECODE_U
+  ldr RS1, PC_ADDRESS
+  sub RS1, RS1, 4
+  add RS1, RS1, IMMEDIATE
+  WRITE_RD(RS1)
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_BEQ:
+  DECODE_S
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  ldr RS2, REGISTER_ADDRESS(RS2)
+  cmp RS1, RS2
+  beq .i_branch_success
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_BGE:
+  DECODE_S
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  ldr RS2, REGISTER_ADDRESS(RS2)
+  cmp RS1, RS2
+  bge .i_branch_success
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_BGEU:
+  DECODE_S
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  ldr RS2, REGISTER_ADDRESS(RS2)
+  cmp RS1, RS2
+  bhs .i_branch_success
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_BLT:
+  DECODE_S
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  ldr RS2, REGISTER_ADDRESS(RS2)
+  cmp RS1, RS2
+  blt .i_branch_success
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_BLTU:
+  DECODE_S
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  ldr RS2, REGISTER_ADDRESS(RS2)
+  cmp RS1, RS2
+  blo .i_branch_success
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_BNE:
+  DECODE_S
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  ldr RS2, REGISTER_ADDRESS(RS2)
+  cmp RS1, RS2
+  bne .i_branch_success
+  NEXT_INST
+.i_branch_success:
+  ldr RS1, PC_ADDRESS
+  and FLAG, FLAG, 0xF
+  lsl FLAG, FLAG, 1
+  sub RS1, RS1, FLAG
+  add RS1, RS1, IMMEDIATE
+  str RS1, PC_ADDRESS
+  b .prepare_trace
+.CKB_VM_ASM_LABEL_OP_DIV:
+  DECODE_R
+  mov TEMP1, INT64_MIN
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  ldr RS2, REGISTER_ADDRESS(RS2)
+  cmp RS1, TEMP1
+  bne .div_branch1
+  cmp RS2, -1
```

Comment:
Not a bug, but we may be able to cut the `INT_MIN/-1`  branch: 

https://developer.arm.com/documentation/den0024/a/The-A64-instruction-set/Data-processing-instructions/Multiply-and-divide-instructions

```
Overflow can only occur in SDIV:
INT_MIN / -1 returns INT_MIN, where INT_MIN is the smallest negative number that can be encoded in the registers used for the operation. The result is always rounded towards zero, as in most C/C++ dialects.
```